PatternPal/PatternPal.Extension/Commands/SubscribeEvents.cs

```diff
@@ -256,13 +259,27 @@ private static void OnDebugProgram(dbgEventReason reason)
             LogEventResponse response = PushLog(request);
         }
 
+        /// <summary>
+        /// The event handler for handling the File.Edit Event. This fires at every save of a document regardless whether a change
+        /// has been made to the file or not. This remains File.Edit, and not File.Save as during the handling of diffs,
+        /// it is determined whether an edit has been made and if so, it is sent with the codebase - therefore: File.Edit.
+        /// </summary>
+        private static void OnDocumentSaved(Document document)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+            LogEventRequest request = CreateStandardLog();
+            request.EventType = EventType.EvtFileEdit;
+            request.CodeStateSection = GetRelativePath(Path.GetDirectoryName(document.FullName), document.FullName);
+            LogEventResponse response = PushLog(request);
+        }
 
         /// <summary>
         /// The event handler for when recognizing software patterns in the extension.
         /// </summary>
-        public static void OnPatternRecognized(RecognizeRequest recognizeRequest, IList<RecognizeResult> recognizeResults)
+        public static void OnPatternRecognized(RecognizeRequest recognizeRequest,
+            IList<RecognizeResult> recognizeResults)
         {
-            if (_package == null || !_package.DoLogData) return; 
+            if (_package == null || !_package.DoLogData) return;
```

Comment:
Isn't this caught in different places for the other events?

Reply:
No, this method is called from the detector view which is initialized earlier than the package. This was also already on develop.